src/auth/plugin.js

```diff
@@ -6,10 +6,10 @@ function addGlobalGuard(router, auth, loginPage) {
 		loginPage = "/login"
 	}
 
-	router.beforeEach((to, from, next) => {
-		// check for login token in local storage
+	router.beforeEach(async (to, from, next) => {
+		// check for existing session
 		if (!auth.loggedIn) {
-			auth.localLogin()
+			await auth.resumeSession()
```

Comment:
On initial page load for a potentially-logged-in user (i.e. we might have an existing session but need to poll the back-end to find out for sure), this causes a tiny moment where the user is shown as not logged-in and router-view is empty, which annoys me slightly.

Reply:
there should be a loading indicator, until the session information is known

Reply:
I added a loading indicator to the user dropdown button. 

I also tried using a loading indicator for the page content until the session is loaded, but I wasn't sure it was better than without the indicator (especially in the datasets page that liked to freeze animations until the dataset table is finished), so I left it out for now.